components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger20.java

```diff
@@ -308,7 +308,18 @@ public String generateAPIDefinition(API api) throws APIManagementException {
             String httpVerb = uriTemplate.getHTTPVerb();
             if (pathItemObject != null) {
                 operationObject = new JSONObject();
-                operationObject.put(APIConstants.SWAGGER_X_AUTH_TYPE, uriTemplate.getAuthType());
+                //Handle auth type specially as swagger need to show exact value
+                String authType = uriTemplate.getAuthType();
+                if (authType.equals(APIConstants.AUTH_APPLICATION_OR_USER_LEVEL_TOKEN)) {
```

Comment:
Code improved in the commit 4ba50436f860ee58034120969504f0a2374e5d62
Thanks @harsha89 for pointing out :+1: